sys/xtimer/xtimer_core.c

```diff
@@ -149,8 +123,27 @@ void _xtimer_set(xtimer_t *timer, uint32_t offset)
         _shoot(timer);
     }
     else {
-        uint32_t target = _xtimer_now() + offset;
-        _xtimer_set_absolute(timer, target);
+        /* time sensitive from "now" access to hardware timer set */
+        uint8_t state = irq_disable();
+        timer->start_time = _xtimer_now();
+        timer->offset = offset;
+        timer->long_offset = 0;
+        timer->target = timer->start_time + offset;
```

Comment:
Could this part be factored out? (at least 4 lines equal to what is done in `_xtimer_set_64() `)

Reply:
We can actually place everything in _xtimer_set_64() to save some coding space. Do you prefer this?